app/assets/src/components/views/amr_heatmap/AMRHeatmapView.jsx

```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { StickyContainer, Sticky } from "react-sticky";
+
+import AMRHeatmapControls from "~/components/views/amr_heatmap/AMRHeatmapControls";
+import AMRHeatmapVis from "~/components/views/amr_heatmap/AMRHeatmapVis";
+import ErrorBoundary from "~/components/ErrorBoundary";
+import { getAMRCounts } from "~/api/amr";
+import { ViewHeader, NarrowContainer } from "~/components/layout";
+
+import cs from "./amr_heatmap_view.scss";
+
+const METRICS = [
+  { text: "Coverage", value: "coverage" },
+  { text: "Depth", value: "depth" },
+];
+
+const VIEW_LEVELS = [
+  { text: "Genes", value: "gene" },
+  { text: "Alleles", value: "allele" },
+];
+
+export default class AMRHeatmapView extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: true,
+      selectedOptions: {
+        metric: "coverage",
+        viewLevel: "gene",
+      },
+    };
+  }
+
+  componentDidMount() {
+    this.requestAMRCountsData(this.props.sampleIds);
+  }
+
+  async requestAMRCountsData(sampleIds) {
+    const rawSampleData = await getAMRCounts(sampleIds);
+    const samplesWithAMRCounts = rawSampleData.filter(
+      sampleData => sampleData.error === ""
+    );
+    this.setState({
+      rawSampleData,
+      samplesWithAMRCounts,
+      sampleIds,
+      loading: false,
+    });
+  }
+
+  assembleControlOptions() {
+    // A Map is used here to preserve the order that the filters will be
+    // presented in the controls element
+    return new Map([
```

Comment:
I was excited in a different comment to learn that Maps have ordered keys.

Still, I think it might be better to use an Array of objects here. You can just store the name of the option as a field in the object. No reason to use a map:
```
[
  {
    key: "viewLevel",
    options: VIEW_LEVELS,
    label: "View Level"
  },
  ...
]
```

A Map also works, but I don't think we use it anywhere else in our code base. I prefer not to introduce new ways of doing things unless we need them.

Reply:
This is a more elegant solution, thank you!